extensions/qbittorrent/src/torrents.tsx

```diff
@@ -0,0 +1,108 @@
+  [RawTorrentState.uploading]: "../assets/uploading.svg",
+  [RawTorrentState.error]: "../assets/error.svg",
+  [RawTorrentState.missingFiles]: "../assets/error.svg",
+  [RawTorrentState.pausedUP]: "../assets/stalledUP.svg",
+  [RawTorrentState.queuedUP]: "../assets/stalledUP.svg",
+  [RawTorrentState.stalledUP]: "../assets/stalledUP.svg",
+  [RawTorrentState.checkingUP]: "../assets/stalledUP.svg",
+  [RawTorrentState.forcedUP]: "../assets/stalledUP.svg",
+  [RawTorrentState.allocating]: "../assets/checking.svg",
+  [RawTorrentState.metaDL]: "../assets/downloading.svg",
+  [RawTorrentState.pausedDL]: "../assets/stalledDL.svg",
+  [RawTorrentState.queuedDL]: "../assets/stalledDL.svg",
+  [RawTorrentState.stalledDL]: "../assets/stalledDL.svg",
+  [RawTorrentState.checkingDL]: "../assets/stalledDL.svg",
+  [RawTorrentState.forcedDL]: "../assets/stalledDL.svg",
+  [RawTorrentState.checkingResumeData]: "../assets/checking.svg",
+  [RawTorrentState.moving]: "../assets/checking.svg",
+  [RawTorrentState.unknown]: undefined,
+};
+
+export default function Command() {
+  const [filter, setFilter] = useState<RawTorrentListFilter>();
+  const [torrents, setTorrents] = useState<RawTorrent[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [updateTimestap, setUpdateTimestap] = useState(+new Date());
+  let updateTimeout: NodeJS.Timeout;
+
+  const { address, username, password, timeout } = getPreferenceValues<Preferences>();
+
+  const qbit = new QBittorrent(address);
+
+  const updateTorrents = async () => {
+    +timeout && updateTimeout && clearTimeout(updateTimeout);
+    setLoading(true);
+    try {
+      await qbit.login(username, password);
+      const torrents = await qbit.api.getTorrents({ filter });
+      setTorrents(torrents);
+    } catch (error) {
+      await showToast({
+        style: Toast.Style.Failure,
+        title: "Failed to load torrents",
+        message: "Please check your preferences.",
+      });
+    }
+    setLoading(false);
+    if (+timeout) {
+      updateTimeout = setTimeout(() => {
+        setUpdateTimestap(+new Date());
+      }, +timeout);
+    }
+  };
+
+  useEffect(() => {
+    updateTorrents();
+  }, [updateTimestap, filter]);
+
+  return (
+    <List
+      isLoading={loading}
+      enableFiltering
+      searchBarPlaceholder="Search your torrents"
+      searchBarAccessory={
+        <List.Dropdown
+          value={filter}
+          tooltip="Filter by state"
+          onChange={(newFilter) => {
+            setFilter(newFilter as any);
+          }}
+        >
+          {Object.keys(RawTorrentListFilter).map((key) => (
+            <List.Dropdown.Item title={key} value={key} key={key} />
+          ))}
+        </List.Dropdown>
+      }
+    >
+      {torrents.map((torrent, index) => {
+        return (
+          <List.Item
+            icon={iconMap[torrent.state]}
+            title={`${index + 1}.${torrent.name} (${prettySize(torrent.size)})`}
```

Comment:
Why do we prefix with the index number here?

Reply:
emmm... I usually want to see how many torrents are downloading, and it might be convenient with index😆😆

Reply:
You could also hold down ⌘ 😄 If you really want to show it you might want to add a space after the index, maybe like this: `title={`#${index + 1} ${torrent.name} (${prettySize(torrent.size)})`}`

Reply:
hold ⌘ sound better. lets remove the index🤔